table.go

```diff
@@ -1384,44 +1375,11 @@ func (t *TableBlock) Serialize(writer io.Writer) error {
 	defer t.table.config.schema.PutWriter(w)
 	defer w.Close()
 
-	// Iterate over all the row groups, and write them to storage
-	count := 0
-	maxRows := 8192
-	j := 0
-	for i, rg := range rowGroups {
-		count += int(rg.NumRows())
-		if count < maxRows { // count can exceed maxRows
-			continue
-		}
-
-		err := t.writeRows(w, count, rowGroups[j:i])
-		if err != nil {
-			return err
-		}
-		j = i
-		count = 0
-	}
-	if count != 0 {
-		err := t.writeRows(w, count, rowGroups[j:])
-		if err != nil {
-			return err
-		}
-	}
-
-	return nil
-}
-
-// writeRows writes a set of dynamic row groups to a writer.
-func (t *TableBlock) writeRows(w *dynparquet.PooledWriter, count int, rowGroups []dynparquet.DynamicRowGroup) error {
-	merged, err := t.table.config.schema.MergeDynamicRowGroups(rowGroups)
-	if err != nil {
-		return err
-	}
-
 	rows := merged.Rows()
 	defer rows.Close()
+	total := 0
 	for {
-		rowsBuf := make([]parquet.Row, count)
+		rowsBuf := make([]parquet.Row, 1)
```

Comment:
I think we should increase the size of this read slice buffer so we can read many rows at a time instead of just 1. Row-by-row seems like it'll be unnecessarily slow.

Reply:
Agreed. We actually already know how many we want so we could make this buffer exactly `rowGroupSize` large.

Reply:
addressing in: https://github.com/polarsignals/frostdb/pull/215